cli/src/constants.rs

```diff
@@ -1,113 +1,224 @@
-pub const MODES: [&str; 2] = ["debug", "release"];
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum Mode {
+    Debug,
+    Release,
+}
 
-//
-// Architeture Section
-//
+pub struct ModeInfo {
+    pub mode: Mode,
+    pub str: &'static str,
+}
 
-// Note that *_ARCH should align with `ios.rs` and `andriod.rs` in "mopro-ffi/src/app_config"
-pub const IOS_ARCHS: [&str; 3] = [
-    "aarch64-apple-ios",
-    "aarch64-apple-ios-sim",
-    "x86_64-apple-ios",
+pub const MODES: [ModeInfo; 2] = [
+    ModeInfo {
+        mode: Mode::Debug,
+        str: "debug",
+    },
+    ModeInfo {
+        mode: Mode::Release,
+        str: "release",
+    },
 ];
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+impl Mode {
+    pub fn as_str(&self) -> &'static str {
+        MODES
+            .iter()
+            .find(|info| info.mode == *self)
+            .map(|info| info.str)
+            .expect("Unsupported Mode, only support 'release' and 'debug'")
+    }
 
+    pub fn parse_from_str(s: &str) -> Self {
+        MODES
+            .iter()
+            .find(|info| info.str.to_lowercase() == s.to_lowercase())
```

Comment:
Want to raise a discussion about string comparison. In normal cases, we usuall convert strings to lower/upper case and then compare them. I'm not sure if that's the case here. All the `parse_from_str` implemeted in this file, I did the same thing.